Remotion/Web/Core/ExecutionEngine/UrlMapping/BaseDirectoryBasedUrlMappingFileFinder.cs

```diff
@@ -0,0 +1,63 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Remotion.FunctionalProgramming;
+using Remotion.ServiceLocation;
+using Remotion.Utilities;
+
+namespace Remotion.Web.ExecutionEngine.UrlMapping
+{
+  /// <summary>
+  /// Base class for <see cref="IUrlMappingFileFinder"/> implementations that want to search for a single URL mapping file.
+  /// </summary>
+  public class BaseDirectoryBasedUrlMappingFileFinder : IUrlMappingFileFinder
+  {
+    /// <summary>
+    /// Creates a new instance using the ambient <see cref="IAppContextProvider"/> and the specified <paramref name="urlMappingFile"/>.
+    /// </summary>
+    public static BaseDirectoryBasedUrlMappingFileFinder Create (string urlMappingFile)
```

Comment:
I see what you did there. Unexpected choice. Can't find a reason to not do it that way. Suggest tweaking the class summary. "base class" implies abtract class and a suffix `Base` in the class name.